src/components/SearchBox/SearchBox.component.jsx

```diff
@@ -32,14 +34,18 @@ const TextBox = styled.input`
   }
 `;
 
-function SearchBox(props) {
+function SearchBox() {
+  const [state, dispatch] = useContext(StateContext);
   const [value, setValue] = useState("");
   const history = useHistory();
   const ENTER_KEY = 13;
 
   function searchIt(e) {
     let search = e.target.value.trim()
-    props.handler(search);
+    if (state.query !== search) {
+      dispatch({ type: types.SEARCH, term: search });

```

Comment:
I recommend moving all the `dispatch` calls to live inside `StateContext` so that the components that use this context don't need to remember the `types` and specific terms for each action like `term`.  So that instead of `dispatch` we could have a function like 
```js
let searchTerm = e.target.value.trim();
if (state.query !== search) {
search(searchTerm);
```